pkg/yurttunnel/util/util.go

```diff
@@ -33,3 +50,43 @@ func RunMetaServer(addr string) {
 		klog.InfoS("meta server stopped listening", "server endpoint", addr)
 	}()
 }
+
+// GetConfiguredDnatPorts returns the DNAT ports configured for tunnel server.
+// NOTE: We only allow user to add dnat rule that uses insecure port as the destination port currently.
+func GetConfiguredDnatPorts(client clientset.Interface, insecurePort string) []string {
+	ports := make([]string, 0)
+	c, err := client.CoreV1().
+		ConfigMaps(YurttunnelServerDnatConfigMapNs).
+		Get(YurttunnelServerDnatConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof("configmap %s/%s is not found",
+				YurttunnelServerDnatConfigMapNs,
+				YurttunnelServerDnatConfigMapName)
+		} else {
+			klog.Errorf("fail to get configmap %s/%s: %v",
+				YurttunnelServerDnatConfigMapNs,
+				YurttunnelServerDnatConfigMapName, err)
```

Comment:
I think that we need to add error return value for func `GetConfiguredDnatPorts`, and when we failed to get yurt-server-tunnel-cfg configmap, we can return error instead of empty slice.